demo/nodejs/mic.js

```diff
@@ -85,18 +87,27 @@ async function micDemo() {
   process.stdin.setRawMode(true);
 
   process.stdin.on("keypress", (key, str) => {
-    if (str.name === 'return') {
+    if (str.sequence === '\r' || str.sequence === '\n') {
       isInterrupted = true;
     }
   });
 
   while (!isInterrupted) {
     const pcm = await recorder.read();
-    const [partialTranscript, isEndpoint] = engineInstance.process(pcm);
-    process.stdout.write(partialTranscript);
-    if (isEndpoint === true) {
-      const finalTranscript = engineInstance.flush();
-      process.stdout.write(`${finalTranscript} `);
+    try {
+      const [partialTranscript, isEndpoint] = engineInstance.process(pcm);
+      process.stdout.write(partialTranscript);
+      if (isEndpoint === true) {
+        const finalTranscript = engineInstance.flush();
+        process.stdout.write(`${finalTranscript} `);
```

Comment:
I think its fine if we write next like after flush? If I'm not wrong this is how we do it in order terminal based demos